Lesson17/ecenaz/contactBook.js

```diff
@@ -0,0 +1,235 @@
+    email: newEmail,
+  });
+  const contactListLengthNew = contactList.length;
+
+  if (contactListLengthPrev < contactListLengthNew) {
+    console.log("Contact added successfully.");
+  } else {
+    console.log("An error occured when adding a contact.");
+  }
+
+  // https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator
+  // contactListLengthPrev < contactListLengthNew
+  //   ? console.log("Contact added successfully.")
+  //   : console.log("An error occured when adding a contact.");
+}
+
+console.log("\nAdding a new contact: Charlie");
+addContact("Charlie", "777-777-7777", "charlie@example.com", contacts);
+displayAllContacts(contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 4: View a Contact by Name
+-----------------------------------------------------------
+Function: viewContact(name)
+- Loops over 'contacts' to find one matching 'name'.
+- Logs the contact info if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function viewContact(name, contactList) {
+  let isContactFound = false;
+  for (const contact of contactList) {
+    if (name === contact.name) {
+      console.log(
+        `name: ${contact.name}, phone:${contact.phone}, email:${contact.email}`,
+      );
+      isContactFound = true;
+    }
+  }
+  if (!isContactFound) {
+    console.log(`No contact found with the name: ${name}`);
+  }
+}
+console.log("\nViewing Bob's contact:");
+viewContact("Bob", contacts);
+
+/*
+-----------------------------------------------------------
+  STEP 5: Update a Contact
+-----------------------------------------------------------
+Function: updateContact(name, newPhone, newEmail)
+- Finds the contact by name and updates phone + email.
+- Logs "Contact updated successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+
+function updateContact(name, newPhone, newEmail, contactList, findContact) {
+  const isContactFound = findContact(contacts, contactList);
+
+  for (const contact of contactList) {
+    if (contact.name === name) {
+      contact.phone = newPhone; contact.email = newEmail;
+
+      console.log(`Contact updated successfullyy: ${name}`);
+      return;
+    }
+  }
+
+  if (!isContactFound) {
+    console.log(`No contact found with the name: ${name}`);
+    return;
+  }
+
+}
+// console.log("\nUpdating Bob's contact:");
+// updateContact("Bob", "999-999-9999", "bob@updated.com");
+// viewContact("Bob");
+
+/*
+-----------------------------------------------------------
+  STEP 6: Remove a Contact
+-----------------------------------------------------------
+Function: removeContact(name)
+- Finds the index of the contact with 'name' using 
+  findIndex() or a loop.
+- Splices it from the array if found.
+- Logs "Contact removed successfully." if found.
+- Otherwise, logs: "No contact found with the name: <name>"
+*/
+function removeContact(name, contacts, addContact, findContact) {
```

Comment:
You don't need to pass addContact and findContact functions as parameters to this function
```suggestion
function removeContact(name, contacts) {
```